src/api/crypto/blockchain/Imp.cpp

```diff
@@ -1077,18 +1068,14 @@ auto Blockchain::Imp::p2wpkh(
 {
     try {
         const auto& hrp = hrp_map_.at(chain);
-        const auto prog = [&] {
-            auto out = UnallocatedVector<std::uint8_t>{};
-            std::transform(
-                hash.begin(),
-                hash.end(),
-                std::back_inserter(out),
-                [](const auto& byte) {
-                    return std::to_integer<std::uint8_t>(byte);
-                });
-
-            return out;
-        }();
+        auto prog = UnallocatedVector<std::uint8_t>{};
+        std::transform(
+            hash.begin(),
+            hash.end(),
+            std::back_inserter(prog),
+            [](const auto& byte) {
+                return std::to_integer<std::uint8_t>(byte);
+            });
```

Comment:
I think we should call `reserve` on the `prog` before `std::transform` to avoid memory reallocations:
`prog.reserve(hash.Blocks().size());`

(If I'm correct and `const Data& hash` is really an object of a class defined in `include/opentxs/network/p2p/Data.hpp`)

Reply:
Yeah sure, but please focus on task itself. I do not want to end up chaning too many things in this task which will be huge anyway:D

Reply:
It's only one line, you can even copy paste it :)

Reply:
I know, and I fully agree that this is good tip, but there are some other places which I could improve but I want to make those PRs as small as possible

Reply:
Sure, so maybe we could create a separate task for this.

Reply:
this one I implemented but my comment was more about future PRs:D there are some more incomming